package/src/versions/resolved-spec.ts

```diff
@@ -0,0 +1,93 @@
+import { SemVer } from "semver";
+import { cleanVersionString, isAliasName } from "./utils";
+import { UnresolvedVersionSpec } from "./unresolved-spec";
+
+// TODO: Maybe make the SemVer library an internal detail and hide it as it
+// would be nice to later switch it for a tree-shakeable alternative.
```

Comment:
ESBuild struggles (or completely don't support?) tree-shaking CJS packages, and the semver lib is quite big. I haven't looked much into alternatives, but there are some out there.